fix: properly handle CLI arguments

leon-richardt opened the pull request:
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Fix CLI arguments not being respected. This happened due to the addition of category-based logging (`--help`, `--version`) and changes to the window loading ( `--channels`), respectively.

When handling `--channels`, I took the liberty to refactor the previous version of window description (which relied on generating JSON) to directly building the `WindowLayout`.

Closes https://github.com/Chatterino/chatterino2/issues/2190.

zneix commented:
Thanks for fixing my laziness

zneix reviewed (approved):
Everything seems to be good, though, you may add this PR's number to the changelog entry about `--channels` parameter.

leon-richardt commented:
> Everything seems to be good, though, you may add this PR's number to the changelog entry about `--channels` parameter.

ACK, added it. I will keep the separate entry in the changelog as well since `--version` and `--help` were also fixed

fanway commented on src/singletons/WindowManager.cpp:
```diff
@@ -297,18 +297,29 @@ void WindowManager::initialize(Settings &settings, Paths &paths)
+
+        if (getArgs().customChannelLayout)
+        {
+            // We would love to use std::optional<T>::value() instead but macOS
+            // does not support it yet. :)
```
But it does though :) (from macOS 10.14 (apple clang 10.0)). Right now chatterino min macOS version is 10.13 for some reason, we could just bump it to 10.14 or even 10.15 :)

leon-richardt replied:
Although I agree an argument could be made to explicitly mention in the comment that this refers to 10.13

fanway replied:
It's not actually about macos version, it's about apple clang version. So you could just use normal compiler and you will be fine :) Also, i think a good point to raise version is that some of the external libs (for example boost) already build for 10.15 and it always gives a warning that it's linked to a different version. Sorry for write it here, kind of offtopic

leon-richardt replied:
> It's not actually about macos version, it's about apple clang version.

Aha, I am little confused about that since the error message explicitly mentions 10.14:
```
../src/singletons/WindowManager.cpp:304:58: error: 'value' is unavailable: introduced in macOS 10.14
```

But then again, I am just vaguely familiar with the Apple ecosystem so I don't feel like I can contribute much to the conversation

fanway replied:
Ye, i agree, message is definitely confusing 😄 Sorry, i just like `::value` usage here, so i thought you could just bump macOS min version :)

leon-richardt replied:
I would prefer to use `::value()` as well but it's not my decision to make :man_shrugging: Would love to hear input from @fourtf and @pajlada on this

leon-richardt replied:
Decided to go with `boost::optional` for now, as suggested by @pajlada on stream.
